### DISCUSSION TITLE
Run javascript after table update (such as sorting)

### DISCUSSION THREAD

**Discussion by scottw-finao** [filamentphp/filament#14367] [Help]
### Package

Table builder

### Package Version

v3

### How can we help you?

I need to run a javascript function after filament/livewire changes a table such as with a filter or a resort.  I have a third party tool that modifies certain links with visual tooltips, but they vanish as soon as the table order or content is modified by filament. I have the name of the function that refreshes the link tooltips, but need some way to run it when the table view is re-built.

**Comment by leandrocfe** [comment]
Could you share some code that you have tried? You can use [this](https://livewire.laravel.com/docs/actions#evaluating-one-off-javascript-expressions) but I need to see what you are trying to achieve

**Comment by scottw-finao** [comment]
dunno what i could show that would be anything relevant to the issue. It's not a form, it's a table. When I click the up/down arrows at the top or type in a search, it re-arrangest the list of items shown.

The code I am using to add tooltips to specific links runs once when the page is finished loading. Which means when filament re-draws the contents of the page, the tooltips are replaced with whatever filament/livewire renders.

I was able to find that `Livewire.hook('morph.updated', ({ el, component }) => { })` does what I need, but I still have an issue as to where 'best' to add the listener. All of the examples I find are saying to use livewire.load event, but when I created a function to just do a console log on that event, it never showed up. In the meantime, I used a page 'onload' with a setTimeout call of 2 seconds to add the listener to make sure livewire is fully loaded and done doing whatever it is doing. This works for the time being, but probably isn't the most eloquent way to do it.

Obviously by using morph.updated, it triggers for every component updated. So I also added a wrapper to it with a short setTimeout that clears itself if called more than once within the timeout period to make sure it only runs one time per morph update.

```
    const whTooltips = {colorLinks: true, iconizeLinks: true, renameLinks: true};

    var rooTimeout = null;

    function refreshOnlyOnce() {
        if(rooTimeout)
            clearTimeout(rooTimeout);
        rooTimeout = setTimeout(doRefresh, 300);
    }

    function doRefresh() {
        WH.Tooltips.refreshLinks();
    }

    window.onload = function() {
        setTimeout(function() {
            Livewire.hook('morph.updated', ({ el, component }) => {
                refreshOnlyOnce()
            })
        }, 1500)
    };
```
